api/proto/teleport/legacy/types/types.proto

```diff
@@ -5306,44 +5306,38 @@ message OktaAssignmentSpecV1 {
     (gogoproto.stdtime) = true,
     (gogoproto.nullable) = true
   ];
-}
 
-// OktaAssignmentActionV1 is a Okta assignment action.
-message OktaAssignmentActionV1 {
-  // OktaAssignmentStatus represents the status of an action in an Okta assignment.
-  enum OktaAssignmentActionStatus {
+  // Status is the status of the assignment.
+  OktaAssignmentStatus status = 4 [(gogoproto.jsontag) = "status"];
+
+  // OktaAssignmentStatus represents the status of an Okta assignment.
+  enum OktaAssignmentStatus {
     // UNKNOWN indicates the status is not set.
     UNKNOWN = 0;
     // PENDING indicates the action has not yet been applied.
     PENDING = 1;
+    // PROCESSSING indicates that the assignment is being applied.
+    PROCESSING = 2;
     // SUCCESSFUL indicates the action was applied successfully.
-    SUCCESSFUL = 2;
+    SUCCESSFUL = 3;
     // FAILED indicates the action was not applied successfully. It will be retried.
-    FAILED = 3;
-    // CLEANED_UP indicates the action was cleaned up successfully.
-    CLEANED_UP = 4;
-    // CLEANUP_FAILED indicates the action was not cleaned up successfully. It will not be retried.
-    CLEANUP_FAILED = 5;
-    // PROCESSSING indicates that the assignment is being applied.
-    PROCESSING = 6;
-    // CLEANUP_PENDING indicates the action needs to be cleaned up.
-    CLEANUP_PENDING = 7;
-    // CLEANUP_PROCESSING indicates the action is being cleaned up.
-    CLEANUP_PROCESSING = 8;
+    FAILED = 4;
   }
 
-  // Status is the status of the action.
-  OktaAssignmentActionStatus status = 1 [(gogoproto.jsontag) = "status"];
-  // Target is the target of the assignment action.
-  OktaAssignmentActionTargetV1 target = 2 [(gogoproto.jsontag) = "target"];
   // LastTransition is an optional field that notes when the last state transition
   // occurred for this action. If absent, this object has never transitioned.
-  google.protobuf.Timestamp LastTransition = 3 [
+  google.protobuf.Timestamp LastTransition = 5 [
     (gogoproto.stdtime) = true,
     (gogoproto.nullable) = false
   ];
 }
 
+// OktaAssignmentActionV1 is a Okta assignment action.
+message OktaAssignmentActionV1 {
+  // Target is the target of the assignment action.
+  OktaAssignmentActionTargetV1 target = 2 [(gogoproto.jsontag) = "target"];
```

Comment:
Do you think we might expand this in the future? Why keep a list of actions that only contain a target instead of just a list of targets?

Reply:
I think I was trying to future proof this by making this very generic, but I think you're right. While I'm doing this surgery I might as well fix this as well.